Makefile

```
@@ -2,15 +2,66 @@ ORG ?= spacemeshos
 IMAGE ?= poet
 BINARY := poet
 
+all: install build
+.PHONY: all
 
-ifdef TRAVIS_BRANCH
-        BRANCH := $(TRAVIS_BRANCH)
-else
-        BRANCH := $(shell git rev-parse --abbrev-ref HEAD)
-endif
+test:
+	gotestsum -- -timeout 5m -p 1 ./...
+.PHONY: test
 
-all: build
-.PHONY: all
+install:
+	go mod download
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.50.0
+	go install github.com/spacemeshos/go-scale/scalegen@v1.0.0
+	go install gotest.tools/gotestsum@v1.8.2
+	go install honnef.co/go/tools/cmd/staticcheck@latest
+.PHONY: install
+
+tidy:
+	go mod tidy
+.PHONY: tidy
+
+test-tidy:
+	# Working directory must be clean, or this test would be destructive
+	git diff --quiet || (echo "\033[0;31mWorking directory not clean!\033[0m" && git --no-pager diff && exit 1)
+	# We expect `go mod tidy` not to change anything, the test should fail otherwise
+	make tidy
+	git diff --exit-code || (git --no-pager diff && git checkout . && exit 1)
+.PHONY: test-tidy
+
+test-fmt:
+	git diff --quiet || (echo "\033[0;31mWorking directory not clean!\033[0m" && git --no-pager diff && exit 1)
+	# We expect `go fmt` not to change anything, the test should fail otherwise
+	go fmt ./...
+	git diff --exit-code || (git --no-pager diff && git checkout . && exit 1)
```

It should do:
```suggestion
	if [ "$(gofmt -s -l . | wc -l)" -gt 0 ]; then exit 1; fi
```

---

A reasonable change but it if a file isn't properly formatted it doesn't show which file.

---

But does it matter? It's a CI check. If it failed, then run `make format` / `gofmt -w .` and done.  It could actually hint to do so:
```suggestion
	if [ "$(gofmt -s -l . | wc -l)" -gt 0 ]; then echo "Code needs reformatting. Please run 'make format'" && exit 1; fi
```